fluture.js

```diff
@@ -717,14 +717,15 @@
 
   Future.do = function Future$do(f){
     check$do(f);
-    const g = f();
-    check$do$g(g);
-    const next = function Future$do$next(x){
-      const o = g.next(x);
-      check$do$next(o);
-      return o.done ? Future$of(o.value) : o.value.chain(next);
-    };
-    return next();
+    return new FutureClass(function Future$do$fork(rej, res){
+      const g = f();
+      check$do$g(g);
+      return (function Future$do$next(x){
+        const o = g.next(x);
+        check$do$next(o);
+        return o.done ? Future$of(o.value) : o.value.chain(Future$do$next);
+      }()).fork(rej, res);
```

Comment:
The process leading upto this was as follows:

1. I removed the `const next = ` part
2. `return next()` would have to be changed
3. but `function Future$do$fork(){ … }` is a function declaration now, which make me feel *iffy*
4. I realized an IIFE would cause it to be a function expression again, coincidentally; 
![scrot_20160427-114608_310x275](https://cloud.githubusercontent.com/assets/1217745/14848393/b9624bfa-0c6d-11e6-9942-6f9618ca687b.png)

I'm not sure now..